routes/api/v1/users.js

```diff
@@ -6,35 +6,51 @@ var bcrypt = require('bcryptjs');
 router.post("/", function(req,res,next) {
   var email = req.body.email
   var password = req.body.password
-  if (req.body.passwordConfirmation === "") {
-    res.setHeader("Content-Type", "application/json");
-    res.status(409).send(`Must provide password confirmation.`)
-  }
-  else if (req.body.password === req.body.passwordConfirmation) {
-    bcrypt.genSalt(10, function(err,salt) {
-      bcrypt.hash(req.body.password, salt, function(err,hash) {
-        User.create({
-          email: email,
-          password: hash,
-          api_key: Math.random().toString(36).substring(2,8)
-          // the arg for toString is the radix which means what base (binary = 2); 36 includes letters
-          // substring takes the characters from index 2 to 8 not including 8
-        })
-        .then(user => {
-          res.setHeader("Content-Type", "application/json");
-          res.status(201).send(`${email} has been created`);
+  var confirm = req.body.passwordConfirmation
+
+  User.findOne({
+    where: {
+      email: email
+    }
+  }).then(result => {
+    if (result) {
+      res.setHeader("Content-Type", "application/json");
+      res.status(409).json({
+        error: `Email already exists in system.`
+      });
+    }
+    else {
+      if (confirm === "") {
+        res.setHeader("Content-Type", "application/json");
+        res.status(409).send(`Must provide password confirmation.`)
+      }
+      else if (password === confirm) {
+        bcrypt.genSalt(10, function(err,salt) {
+          bcrypt.hash(req.body.password, salt, function(err,hash) {
+            User.create({
+              email: email,
+              password: hash,
+              api_key: Math.random().toString(36).substring(2,8)
+              // the arg for toString is the radix which means what base (binary = 2); 36 includes letters
+              // substring takes the characters from index 2 to 8 not including 8
+            })
```

Comment:
Not sure if you wanted to keep out these commented lines, but thought I would point it out in case you meant to get rid of them.

Reply:
Thanks! I wanted to keep them in for a little bit so I could keep tabs on what I did. I'll probably remove them later.